keras_nlp/layers/position_embedding.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Position embedding implementation based on `keras.layers.Layer`."""
+
+import tensorflow as tf
+from tensorflow import keras
+
+
+class PositionEmbedding(keras.layers.Layer):
+    """Creates a layer which learns a position embedding for inputs sequences.
+
+    This class assumes that in the input tensor, the last dimension corresponds
+    to the features, and the dimension before the last corresponds to the
+    sequence.
+
+    Args:
+      max_length: The maximum length of the dynamic sequence.
+      initializer: The initializer to use for the embedding weights. Defaults to
+        "glorot_uniform".
+      seq_axis: The axis of the input tensor where we add the embeddings.
+
+    Example:
+    ```python
+    class PositionalEmbedding(layers.Layer):
```

Comment:
Let's rewrite this without the class wrapper. Was talking with Francois, I think we may want to add this wrapper you are showing as well as a separate layer `TokenAndPositionEmbedding`.

But this should show using the two layers standalone on some input and summing them.

Reply:
Ok, I'll modify this to remove the class